### PR TITLE
feat: add temporal reasoning test suite (Tier 1, 22 deterministic tests)

### PR DESCRIPTION
**Pull request by tkarcheski**
## Summary

Adds a new `temporal_reasoning` test suite with 22 deterministic Robot Framework tests evaluating LLM time-and-date reasoning — a well-documented weak area for language models. All tests are **Tier 1 / verify:python**: the LLM's response is graded by Python extraction logic, not by another LLM, giving reliable pass/fail signals with zero grader variance.

Research basis: temporal arithmetic (date offsets, duration conversions) and chronological ordering are recurring failure modes in LLM benchmarks (e.g., TimeQA, TempQuestions). The Tier 1 grading approach directly improves test accuracy relative to Tier 2/3 suites that depend on an LLM judge.

## How to review

**Start here:** `src/rfc/temporal_reasoning_keywords.py` — two keywords, two private helpers.

**Key changes:**
1. `src/rfc/temporal_reasoning_keywords.py` — keyword library: `Evaluate Temporal Arithmetic` (extracts first integer, handles comma thousands-separators) and `Evaluate Chronological Order` (extracts A/B from first line with word-boundary guard)
2. `tests/test_temporal_reasoning_keywords.py` — 44 unit tests covering all extraction edge cases (comma formatting, prose embedding, mid-line letter guard, word-boundary checks) and both keyword methods
3. `robot/temporal_reasoning/test_time_arithmetic.robot` — 8 tests: seconds/minutes/hours conversions (e.g., 6h40m → 400 min)
4. `robot/temporal_reasoning/test_calendar_arithmetic.robot` — 8 tests: days-in-month, leap-year, day-of-year offsets
5. `robot/temporal_reasoning/test_event_chronology.robot` — 6 tests: A/B historical event ordering (3 where A is correct, 3 where B is correct)
6. `robot/temporal_reasoning/variables/*.yaml` — scenario data separated from test logic
7. `config/test_suites.yaml` + `config/local_models.yaml` — suite registration

**What to ignore:** The resource file and `__init__.robot` are mechanical boilerplate identical to the causal_reasoning suite pattern.

## Evidence of testing

<details>
<summary>pytest output</summary>

```
2511 passed, 1 skipped, 6 warnings in 10.10s
(44 new tests in test_temporal_reasoning_keywords.py, all passing)
```
</details>

<details>
<summary>pre-commit output</summary>

```
check yaml...............................................................Passed
check json...............................................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
check for merge conflicts................................................Passed
ruff (legacy alias)......................................................Passed
ruff format..............................................................Passed
mypy.....................................................................Passed
Block FTP usage..........................................................Passed
```
</details>

<details>
<summary>code-quality-check output</summary>

```
src/rfc/temporal_reasoning_keywords.py   58      0   100%
...
Total coverage: 87.63%  (above 50% threshold)
All checks passed!
```
</details>

<details>
<summary>robot-dryrun output</summary>

```
609 tests, 597 passed, 12 failed
(12 pre-existing failures from Playwright tests missing browser library;
 unchanged from baseline of 587 tests / 575 passed / 12 failed.
 All 22 new temporal_reasoning tests pass in dry-run.)
```
</details>

## Critical changes

None. No public keyword API changes, no database schema changes, no config format changes. New suite is additive only; it is registered in both `config/test_suites.yaml` and `config/local_models.yaml` so CI and `run_local_models.py` will pick it up automatically.

## Checklist

- [x] All pytest tests pass
- [x] `pre-commit run --all-files` passes
- [x] `make code-quality-check` passes (lint + mypy + coverage)
- [x] `make robot-dryrun` passes for new suite (12 pre-existing Playwright failures unchanged)
- [x] Every new Robot test tagged with exactly one `tier:*` and one `verify:*`
- [x] New suite registered in `config/test_suites.yaml` and `config/local_models.yaml`
- [x] Version bumped to 1.10.3 (patch — additive new suite)
- [x] No stray changes outside intended scope (verified via `git diff --staged`)
- [x] No `uv.lock` committed

https://claude.ai/code/session_01TWUCrGiZTPNo45XNRyxGtk

---
_Generated by [Claude Code](https://claude.ai/code/session_01TWUCrGiZTPNo45XNRyxGtk)_

### CODE REVIEW

File: src/rfc/temporal_reasoning_keywords.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+def _extract_choice_letter(response: str) -> Optional[str]:
+    """Return ``'A'`` or ``'B'`` extracted from the first non-empty line.
+
+    The prompt instructs the model to place the letter first. Returning
+    ``None`` signals a non-compliant response; callers treat that as failure.
+    """
+    if not response:
+        return None
+    first_line = response.strip().splitlines()[0]
```
[the remark this answers is not in the document]
Comment (tkarcheski) [reply src/rfc/temporal_reasoning_keywords.py]: Good catch. Fixed in bb9d4d1.

`_extract_choice_letter` now stores `response.strip()` in `stripped`, returns `None` early if `stripped` is falsy, then indexes into `stripped.splitlines()`. A whitespace-only input like `"\n\n"` correctly returns `None` and is graded as incorrect rather than crashing.

Two regression tests added: `test_returns_none_on_whitespace_only` (`"\n\n"`) and `test_returns_none_on_spaces_only` (`"   "`). All 46 tests pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/rfc/temporal_reasoning_keywords.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+from .llm_client import create_provider, resolve_timeout
+from .rfc_data import emit_rfc_data
+
+# Matches the first standalone integer, tolerating comma thousands-separators
+# (e.g. "3,600" is treated as 3600, not 3).
+_INTEGER_RE = re.compile(r"\b(\d[\d,]*\d|\d)\b")
+
+# Matches "A" or "B" anchored to the start of the first non-empty line.
+# Word boundary prevents matching "Be careful" or "About" as a choice letter.
+_AB_CHOICE_RE = re.compile(r"^\s*([ABab])\b")
```
[the remark this answers is not in the document]
Comment (tkarcheski) [reply src/rfc/temporal_reasoning_keywords.py]: Fixed in 213b410.

`_AB_CHOICE_RE` updated to `^\s*(?:Answer:\s*)?([ABab])\b` with `re.IGNORECASE`. The optional `(?:Answer:\s*)?` group absorbs the echoed label so `"Answer: A\n..."` and bare `"A\n..."` both extract correctly. The word-boundary guard still blocks `"Answer: Be careful"` from matching.

5 regression tests added: uppercase prefix, choice B, lowercase (`answer: a`), no-space colon (`Answer:B`), and prose-after-prefix guard. All 51 tests pass.

---
_Generated by [Claude Code](https://claude.ai/code)_
